annotations/forms.py

```diff
@@ -169,7 +169,37 @@ def label_from_instance(self, obj):
         """
         return obj.uri
 
-
+    def to_python(self, value):
+        if value in self.empty_values:
+            return None
+        try:
+            key = 'uri'
+            py_value = self.queryset.get(**{key: value})
+        except self.queryset.model.DoesNotExist:
```

Comment:
in here the conceptpower class should be used (or whereever the conceptpower logic is implemented). No reason to have another version of code that talks to Conceptpower.